src/ProductConstructionService/ProductConstructionService.DependencyFlow/PullRequestBuilder.cs

```diff
@@ -137,84 +138,98 @@ public async Task<string> CalculatePRDescriptionAndCommitUpdatesAsync(
+                coherencyUpdatesPerDirectory[targetDirectory] = [.. coherencyUpdates];
+                itemsToUpdate.AddRange(coherencyUpdates
+                    .Select(du => du.To));
+            }
 
-            startingReferenceId = await AppendBuildDescriptionAsync(description, startingReferenceId, update, deps, committedFiles, build);
+            if (itemsToUpdate.Count > 0)
+            {
+                targetDirectoryUpdatedDependencies = await remote.GetUpdatesAsync(
+                        targetRepository,
+                        newBranchName,
+                        itemsToUpdate,
+                        targetDirectory);
+                await locationResolver.AddAssetLocationToDependenciesAsync(itemsToUpdate);
+                updatedDependencies.AddRange(targetDirectoryUpdatedDependencies);
+            }
         }
 
-        // If the coherency update wasn't combined, then
-        // add it now
-        if (!combineCoherencyWithNonCoherency && coherencyUpdate.update != null)
+        if (updatedDependencies.Count > 0)
         {
-            var message = new StringBuilder();
-            var build = await _barClient.GetBuildAsync(coherencyUpdate.update.BuildId);
-            await CalculateCommitMessage(coherencyUpdate.update, coherencyUpdate.deps, message);
-            AppendCoherencyUpdateDescription(description, coherencyUpdate.deps);
-
-            var itemsToUpdate = coherencyUpdate.deps
-                .Select(du => du.To)
-                .ToList();
-
-            await locationResolver.AddAssetLocationToDependenciesAsync(itemsToUpdate);
+            if (nonCoherencyCommitMessage.Length > 0)
+            {
+                nonCoherencyCommitMessage.Insert(0, $"Update dependencies from {update.SourceRepo} build {build.AzureDevOpsBuildNumber}" + Environment.NewLine);
+                nonCoherencyCommitMessage.AppendLine();
+            }
+            if (coherencyCommitMessage.Length > 0)
+            {
+                coherencyCommitMessage.Insert(0, "Dependency coherency updates" + Environment.NewLine);
+                coherencyCommitMessage.AppendLine();
+            }
             await remote.CommitUpdatesAsync(
+                updatedDependencies,
                 targetRepository,
                 newBranchName,
-                itemsToUpdate,
-                message.ToString());
-        }
+                nonCoherencyCommitMessage.Append(coherencyCommitMessage).ToString());
 
-        // If the coherency algorithm failed and there are no non-coherency updates and
-        // we create an empty commit that describes an issue.
-        if (requiredUpdates.Count == 0)
+            if (coherencyUpdatesPerDirectory.Count > 0)
+            {
+                AppendCoherencyUpdateDescription(description, coherencyUpdatesPerDirectory);
+            }
+            if (nonCoherencyUpdatesPerDirectory.Count > 0)
+            {
+                await AppendBuildDescriptionAsync(
```

Comment:
Previously, this had `startingReferenceId = ...`. Is it fine that disappeared?

Reply:
Before this was a ref parameter and now it's being returned so if this is called multiplate times, it needs to be caught and reused

Reply:
it's only being called once now tho. It used to be in a loop so we had to reuse it

Reply:
In that case it maybe does not need to return the ID?

Reply:
true, removed